### PR TITLE
Wait for the central manager to be powered on before retrieving the peripheral instance

### LINKED ISSUE #25: Bluetooth transport may be broken on iOS 13
Hi,
On iOS 13 it is now required to wait until the `CBCentralManager` is in `.poweredOn` state before calling `retrievePeripherals(withIdentifier:)`. Have a look at this line: https://github.com/JuulLabs-OSS/mcumgr-ios/blob/13ba851af50d8ca5fdcd89e5b592723b30a857ed/Source/Bluetooth/McuMgrBleTransport.swift#L116-L117

In DFU Library for iOS we had similar issue, which has been fixed with this commit: https://github.com/NordicSemiconductor/IOS-Pods-DFU-Library/commit/a64315b516f5c9c8bb52451b4b277e79a1ec9640
Retrieving peripheral was delayed until the manager was ready. Similar fix should be applied in MCUMGR library.

### PR DESCRIPTION
**Pull request by amorde**
Closes #25

This resolves an issue in iOS 13 - iOS 13 is more strict and will not return a peripheral instance until the central manager is in the `poweredOn` state.

### CODE REVIEW

File: Source/Bluetooth/McuMgrBleTransport.swift [hunk 1]
```diff
@@ -114,25 +114,30 @@ public class McuMgrBleTransport: NSObject {
...
     }
     
     public var name: String? {
-        return peripheral.name
+        return peripheral?.name
     }
     
-    public var identifier: UUID {
-        return peripheral.identifier
+    public var identifier: UUID
```
Comment (amorde) [inline Source/Bluetooth/McuMgrBleTransport.swift]: this probably shouldn't be publicly writable - I'll change it to `public private(set)`

**Comment by amorde** [comment]
Tested on iOS 13.1 beta 2, and iOS 12.4.1

File: Source/Bluetooth/McuMgrBleTransport.swift [hunk 2]
```diff
@@ -200,18 +205,21 @@ extension McuMgrBleTransport: McuMgrTransport {
...
         }
-        
-        // Wait until the Central Manager is ready.
-        // This is required when the manager has just been created.
-        if centralManager.state == .unknown {
+
+        // We might not have a peripheral instance yet, if the Central Manager has not
+        // reported that it is powered on.
+        // Wait until it is ready, and timeout if we do not get a valid peripheral instance
+        var targetPeripheral: CBPeripheral
```
Comment (davidtaylor-juul) [inline Source/Bluetooth/McuMgrBleTransport.swift]: This can be `let`

Comment (amorde) [reply Source/Bluetooth/McuMgrBleTransport.swift]: ah yes, good call

**Pull request merged**